java-checks/src/main/java/org/sonar/java/checks/InterfaceStaticMutableMemberCheck.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012 SonarSource
+ * dev@sonar.codehaus.org
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import org.sonar.check.BelongsToProfile;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.model.declaration.VariableTreeImpl;
+import org.sonar.java.resolve.Type;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.Modifier;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+import java.util.List;
+
+@Rule(
+  key = "S2386",
+  priority = Priority.CRITICAL,
+  tags = {"unpredictable"})
+@BelongsToProfile(title = "Sonar way", priority = Priority.CRITICAL)
+public class InterfaceStaticMutableMemberCheck extends SubscriptionBaseVisitor {
+
+  private static final String MESSAGE = "Move \"%s\" to a class and lower its visibility";
+  private static final String COLLECTION_QUALIFIED_NAME = "java.util.Collection";
+  private static final String DATE_QUALIFIED_NAME = "java.util.Date";
```

Comment:
I would inline the constant here. It just brings an unecessary level of indirection in those cases.